packages/component/src/util.js

```diff
@@ -12,3 +12,33 @@ export function warn(message) {
   // eslint-disable-next-line no-console
   console.warn(`loadable: ${message}`)
 }
+
+export const STATUS_PENDING = 'PENDING'
+export const STATUS_RESOLVED = 'RESOLVED'
+export const STATUS_REJECTED = 'REJECTED'
+
+export function statusAware(promise) {
+  let status = STATUS_PENDING
+
+  const proxy = new Proxy(promise, {
```

Comment:
No `Proxies` please. They are not supported widely enough yet.
And why this needs a recursive wrapping?

Reply:
The recursion is that if you call then or catch, it enables you to access status on that newly created promises from those calls